lib/charms/opensearch/v0/helper_security.py

```diff
@@ -136,58 +138,124 @@ def split_ca_chain(pem_content: str) -> list[str]:
+                run_cmd(
+                    f"{KEYTOOL} -changealias "
+                    f"-alias {ix_alias} -destalias {old_alias} "
+                    f"-keystore {store_path} -storetype PKCS12",
+                    f"-storepass {store_pwd}",
                 )
             except OpenSearchCmdError as e:
-                # This message means there was no "ca" alias or store before, if it happens ignore
-                if not (
-                    e.out is not None
-                    and (
-                        f"Alias <{alias}-{index}> does not exist" in e.out
-                        or "Keystore file does not exist" in e.out
-                    )
-                ):
-                    raise
-
-        with tempfile.NamedTemporaryFile(
-            mode="w+t", dir=os.path.dirname(store_path)
-        ) as ca_tmp_file:
-            ca_tmp_file.write(certs[index])
-            ca_tmp_file.flush()
+                msg = (e.out or "") + (e.err or "")
+                if ("does not exist" not in msg) and ("Keystore file does not exist" not in msg):
+                    return False
+
+        # import the cert
+        fd, tmpfile = tempfile.mkstemp(dir=tmpdir)
+        try:
+            with os.fdopen(fd, "w", encoding="utf-8", errors="replace") as f:
+                f.write(pem)
+
             try:
                 run_cmd(
-                    f"""{KEYTOOL} -importcert \
-                    -noprompt \
-                    -alias {alias}-{index} \
-                    -keystore {store_path} \
-                    -file {ca_tmp_file.name} \
-                    -storetype PKCS12
-                    """,
+                    f"{KEYTOOL} -importcert -noprompt "
+                    f"-alias {ix_alias} -keystore {store_path} -file {tmpfile} -storetype PKCS12",
                     f"-storepass {store_pwd}",
                 )
-                run_cmd(f"sudo chmod +r {store_path}")
-                logger.info("New CA was added to truststore.")
-            except OpenSearchCmdError as e:
-                logger.error("Error storing the ca-cert: %s", e)
-                return False
+            except OpenSearchCmdError:
+                if tolerate_import_if_listed:
+                    listed = list_cas(store_pwd=store_pwd, store_path=store_path) or {}
+                    if ix_alias in listed:
+                        pass
+                    else:
+                        return False
+                else:
+                    return False
+        finally:
+            try:
+                os.remove(tmpfile)
+            except FileNotFoundError:
+                pass
+
+    # post-actions
+    try:
+        if owner:
+            run_cmd(f"sudo chown {owner} {store_path}")
+        if final_mode:
+            run_cmd(f"sudo chmod {final_mode} {store_path}")
+        if add_read_perm:
+            run_cmd(f"sudo chmod +r {store_path}")
+    except OpenSearchCmdError:
+        pass
```

Comment:
Similar comment here, why not always apply? and to avoid one operation raise and make the remaining operations fail, we should perhaps group them:
```suggestion
    try:
        run_cmd(f"sudo chown snap_daemon:root {store_path}; sudo chmod 640 {store_path}")
    except OpenSearchCmdError:
        pass
```

Reply:
we can  group them as for s3_ca_chain we need to change owner and  final_mode but for general flow we need add_read_perm.

Reply:
They are grouped.